apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,365 @@
+import { Meta, Story } from '@storybook/addon-docs';
+import { SlotSketch } from './SlotSketch.stories';
+
+<Meta title="Concepts/Developer/Customizing Components with Slots" />
+
+## Customizing Components with Slots
+
+Fluent UI React components have parts that are designed to be replaced.
+These are called slots.
+Slots provide a more flexible approach over render callbacks used in previous versions.
+Fluent UI slots are conceptually similar to slots in other component libraries and frameworks,
+but there are some differences.
+
+- Each slot is exposed as a top-level prop of the same name.
+- Some slots have default content and others are empty by default.
+- Slots may target different types of elements or components to restrict the type of content.
+- You can fill a slot with a primitive value, JSX/TSX, props objects, or a render function.
+
+You can think of component slots like a template where the dotted line parts can be cut out and replaced.
+For example, the `Input` component has `contentBefore` and `contentAfter` slots.
+
+<SlotSketch />
+
+### When to use slots
+
+Use a component's slots when you want to
+
+- Setting the content of component part
+- customize the style of a component part (via `className`)
+- customize the props passed to a component part
+- subscribe to event handlers of a component part
+- change the element type of a component part (via `as`)
+- completely replace existing content of a slot
+
+Later in this topic, you will find examples covering each of these scenarios.
+
+### When to not to use slots
+
+- If you need to slightly adjust the display of components across you application, prefer to customize the theme.
+  Fluent UI React components leverage theme design tokens to render consistently.
+  For example, if you want to have borders to be a specific color on `:hover`, you can create a theme that sets `colorNeutralStroke1Hover`.
+
+- If you need to slightly adjust the display of a specific component instance, prefer to apply a custom style.
+  For example, if you want to make a particular component have a thicker border.
+  You create the style using `makeStyles` and/or `mergeClasses` and then apply it to the component using `className`.
+
+- If you want to change how a component behaves, make deep style customizations,
+  replace non-slot parts, or wrap a component with different props, then consider using the hooks API.
+  The hooks API gives you complete control to recompose a component but is more complex than using slots.
+
+### Conditional Rendering
+
+Some components conditionally render slots.
+
+For example, Avatar has `label` slot that only renders when there is no image provided.
```

Comment:
Is this a gotcha for users to look out for? i.e. "So if you provide a label and an image, don't expect the label to be rendered"